src/main/java/de/unistuttgart/iste/gits/flashcard_service/service/FlashcardService.java

```diff
@@ -115,5 +117,41 @@ public List<FlashcardSet> getFlashcardSetsByAssessmentId(List<UUID> ids) {
                 .map(flashcardMapper::flashcardSetEntityToDto)
                 .toList();
     }
+    /**
+     * removes all flashcards when linked Content gets deleted
+     *
+     * @param dto event object containing changes to content
+     */
+    public void removeContentIds(ContentChangeEvent dto) {
+
+        // validate event message
+        try {
+            checkCompletenessOfDto(dto);
+        } catch (NullPointerException e) {
+            log.error(e.getMessage());
+            return;
+        }
+        // only consider DELETE Operations
+        if (!dto.getOperation().equals(CrudOperation.DELETE) || dto.getContentIds().isEmpty()) {
+            return;
+        }
+
+        // find all quizzes
+        List<FlashcardEntity> flashcardEntities = flashcardRepository.findAllById(dto.getContentIds());
```

Comment:
This won't work. The content ID matches the FlashcardSET ID, not the flashcard ID